battles/forms.py

```diff
@@ -1,166 +1,162 @@
-            'status': forms.HiddenInput(),
-        }
+        fields = ('trainer_opponent', )
 
     def __init__(self, *args, **kwargs):
-        initial = kwargs.get('initial', {})
-        initial_user_id = initial.get('user') and initial.get('user').id
-        initial['trainer_creator'] = initial.get('trainer_creator', initial_user_id)
-        initial['status'] = initial.get('status', 'ON_GOING')
-        kwargs['initial'] = initial
         super().__init__(*args, **kwargs)
+        users = User.objects.exclude(id=self.initial['trainer_creator'].id)
+        self.fields['trainer_opponent'].queryset = users
 
     def clean(self):
         cleaned_data = super().clean()
-        pokemons = [
-            self.cleaned_data['pokemon_1'].lower().strip(),
-            self.cleaned_data['pokemon_2'].lower().strip(),
-            self.cleaned_data['pokemon_3'].lower().strip()
+        rounds = [
+            int(self.cleaned_data['order_1']),
+            int(self.cleaned_data['order_2']),
+            int(self.cleaned_data['order_3']),
         ]
+        if len(Counter(rounds)) != 3:
+            raise forms.ValidationError(
+                'Trainer, select a different value for each round field.'
+            )
 
-        for pokemon in pokemons:
-            if not pokemon_exists(pokemon):
-                raise forms.ValidationError(
-                    f'We couldnt find "{pokemon}". Please, check if you wrote it correctly.'
-                )
-
+        pokemons = order_battle_pokemons(self.cleaned_data)
         if not is_pokemons_sum_valid(pokemons):
             raise forms.ValidationError(
                 'Trainer, your pokemon team stats can not sum more than 600 points.'
             )
-
         return cleaned_data
 
-    def save(self, commit=True):
-        pokemons = [
-            self.cleaned_data['pokemon_1'].lower().strip(),
-            self.cleaned_data['pokemon_2'].lower().strip(),
-            self.cleaned_data['pokemon_3'].lower().strip()
-        ]
-        self.instance.save()
-
-        for pokemon in pokemons:
-            if Pokemon.objects.filter(name=pokemon).count() == 0:
-                Pokemon.objects.create(
-                    name=pokemon,
-                    attack=get_pokemon_stats(pokemon)['attack'],
-                    defense=get_pokemon_stats(pokemon)['defense'],
-                    hitpoints=get_pokemon_stats(pokemon)['hitpoints'],
-                )
-
-        TrainerTeam.objects.create(
-            trainer=self.initial['user'],
-            pokemon_1=Pokemon.objects.get(name=pokemons[0]),
-            pokemon_2=Pokemon.objects.get(name=pokemons[1]),
-            pokemon_3=Pokemon.objects.get(name=pokemons[2]),
-            battle_related=self.instance,
-        )
-
-        return super().save()
-
 
 class SelectTrainerTeamForm(forms.ModelForm):
-    pokemon_1 = forms.CharField()
-    pokemon_2 = forms.CharField()
-    pokemon_3 = forms.CharField()
+    pokemon_1 = forms.ModelChoiceField(
```

Comment:
@lucabezerra The main differences are the related model and its attributes.

The `CreateBattleForm` has `Battle` as model, because it's the battle initialization, so saves infos as trainer creator and trainer opponent, but also need to save the trainer creator pokemon team.

The `SelectTrainerTeamForm` has `TrainerTeam` as model, because it's the battle challenge response, so just need to save the trainer opponent pokemon team. Infos such  `status` e `trainer winner` from `Battle` model are updated from functions in the `views`.

Reply:
Got it! In this case, what do you think about creating some base form class and inheriting from it in each of these models, so we can avoid this big code repetition? Do you think it's feasible?

Reply:
Done!